main.go

```diff
@@ -85,8 +85,15 @@ func main() {
 		DevMode = false
 	}
 
+	loggerOpts := &zap.Options{
+		Development: DevMode,
```

Comment:
when development mode is set it enables the debug level, and brings logs from [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime)

Reply:
should we make DEV_MODE=false in CI?

Reply:
I think it is fine to keep it `true`. Even though most of the `DEBUG` messages that we have now are not relevant (they are only informing us that new k8s `events` were created), in the future we can/should define the INFO/DEBUG/WARN messages in a better way, which will be helpful in CI.

Another thing that I was thinking about is to check if it is possible to allow users to define (I don't know if through an annotation or CR spec) the log level. Maybe we should open a new issue to check this?